packages/cli/src/ceramic-daemon.ts

```diff
@@ -754,10 +770,54 @@ export class CeramicDaemon {
     }
 
     // Process request
-    await successCallback(jwsValidation.models)
+    switch (successCallback.type) {
```

Comment:
This logic looks weird to me, there's a callback function provided in the method call, but it needs specific logic in this handler?
Couldn't the branching logic be handled by the caller directly rather than being split between different methods?

Reply:
Can you describe what that would look like in the caller?

Reply:
Since we're adjusting the response object in this method, trying to move this functionality out produces a lot of code duplication, and actually makes the code harder to read. Going to leave this as is.